specification/semantic_conventions.md

```diff
@@ -183,3 +185,27 @@ provides a range of lines, the second `lineno` can be provided after a colon `:`
 - ` <col>` or ` <col>:<col>` - OPTIONAL - If the runtime provides a column or column range, it MAY be provided
 after the `lineno`, separated by a space.
 - literal `)`
+
+#### PPROF Profile.proto Data Format
+
+`Profile.proto` is a data representation for profiling data defined at https://github.com/google/pprof/tree/master/proto.
+Log message will contain a gzip-compressed, base64-encoded protocol buffer conforming to `profile.proto`. Each message
+contains either `allocation` or `profiling` samples determined by the data type specified for the log record.
+
+For each `allocation` and `profiling` sample:
+- label `source.event.period` of type `long` MUST contain the sampling period if this sample represents a periodic event
```

Comment:
This is a bit confusing -- IIUC, `allocation` samples do not have this field, as it is only relevant for `profiling`. Also I wonder if we can put it in the value instead of an attribute?

Reply:
Moved it under `profiling` for now. We can make it a value instead of a label if you believe it is better that way.

Reply:
Don't have strong feelings about that, but seems to be more consistent to put both as the "value". That's the approach we take on the backend too. @vovencij wdyt?

Reply:
My take on this is that interpreting `event.period` as value in the agent is not entirely semantically correct. One sample in pprof represents a measurement, which may have additional attributes besides the time the measurement was taken. Number of bytes allocated - this is a value of a measurement, no questions asked. In case of periodic thread call stack sampling, as cpu profiler does, the measurement is basically a count of 1 - the fact that at a particular time one particular call stack was observed for one time. Period at which we take samples - this is not a value, but an additional attribute. Yes, at the backend we do use this attribute as an approximate value when drawing a flamegraph, but this is not a value of that sample at the agent side.